pkg/controllers/proxy/service_endpoints_sync.go

```diff
@@ -861,24 +858,3 @@ func (nsc *NetworkServicesController) cleanupDSRService(fwMark uint32) error {
 	delete(nsc.fwMarkMap, fwMark)
 	return nil
 }
-
-func (nsc *NetworkServicesController) cleanupStaleMetrics(activeServiceEndpointMap map[string][]string) {
```

Comment:
Are you sure that we don't need any sort of cleanup function any longer with the changes you've made? I haven't gotten a chance to test it yet, but have you made sure that when a service is removed it no longer produces stale metrics?

I remember before this function was added there were quite a few issues filed complaining about how it was ruining people's prometheus cardinality by continuing to track stale metrics for services and label sets that were no longer in the cluster.

Reply:
As metrics are collected real-time and created manually, they are not stored anywhere. With this refactor there will be no stale metrics.